app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,50 @@
+class MessagesController < ApplicationController
+
+  before_action :authenticate_user!
+
+  def run_code
+  end
+
+  def new
+    @message = Message.new
+    @template = Template.find_by(id: params[:template_id])
+  end
+
+  def create
+    @message = Message.new(message_params)
+    if @message.save
+      if request.xhr?
+        @users = User.all
+        render "users/_index", locals: {user: @users, message_id: @message.id}, layout: false
+      else
+        @users = User.all
+        redirect_to users_path
+      end
+    else
+      flash.now[:errors]="Oops! Looks like you made a mistake!"
```

Comment:
What if it was xhr and failed? Maybe refactor into two methods to avoid an if/else extravaganza.